patches/tModLoader/Terraria.ModLoader/AltAdamantite.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections.Generic;
+using Terraria.ID;
+
+namespace Terraria.ModLoader
+{
+	public class AltAdamantite
+	{
+		public readonly int oreTileType;
+		public readonly string oreBlessingText;
+		public readonly float abundanceMultiplier;
+		private static readonly List<AltAdamantite> AltAdamantites = new List<AltAdamantite>();
+		internal static bool vanillaAltsAdded = false;
+		public static AltAdamantite ChosenAltAdamantite { get; private set; }
+
+		public AltAdamantite(int oreTileType, string oreBlessingText, float abundanceMultiplier = 1f)
+		{
+			this.oreTileType = TileLoader.TileCount > oreTileType ? oreTileType : throw new Exception("oreTileType is an invalid tile");
+			this.oreBlessingText = oreBlessingText;
+			this.abundanceMultiplier = abundanceMultiplier;
+		}
+
+		public static AltAdamantite GetAltAdamantite()
```

Comment:
Similarly to above, I feel like this should be abstracted. E.g: AltOre class, GetAltOre()